pkg/tfbridge/names.go

```diff
@@ -49,19 +49,28 @@ func PulumiToTerraformName(name string, tfs map[string]*schema.Schema) string {
 		if ok && sch.MaxItems != 1 && (sch.Type == schema.TypeList || sch.Type == schema.TypeSet) {
 			result = singularResult
 		}
+
+		if ps != nil {
+			p, ok := ps[singularResult]
+			if ok && p != nil && p.MaxItemsOne != nil && !*p.MaxItemsOne &&
+				(sch.Type == schema.TypeList || sch.Type == schema.TypeSet) {
+				result = singularResult
+			}
+		}
```

Comment:
Just to make this logic a little clearer, and to avoid duplicating a bunch of this - perhaps compute “isMaxItemsOne” first using both schema and schemainfo, then have just one test for the condition that would choose the singular result?

Reply:
done!